### PR TITLE
Notifications Page

### PR DESCRIPTION
**Pull request by zackaidja**
Added a Notifications page that alerts users during different situations, these situations are currently:
1. When a user requests to join an event you created.
2. When your request to join an an event is accepted
3. When your request to join an an event is rejected
4. When you have been removed from an event list

In the future we can add more functionality to the notifications page for things like friend requests and comments on event posts and replies once that functionality is added

### CODE REVIEW

File: events/views.py [hunk 1]
```diff
@@ -371,8 +371,18 @@ def toggleJoinRequest(request, event_id):
...
+                message=f"{request.user.username} withdrew their request to join the event '{event.id}'.",
+            )
         else:
             join.status = PENDING
+            # Create a notification when a new request is submitted
+            Notification.objects.create(
+                user=event.creator,
+                message=f"{request.user.username} has requested to join the event '{event.id}'.",
+            )
         join.save()
```
Comment (WenjunYuAnny) [inline events/views.py:386]: When I tested locally, it appears that when a user requests to join, the notification doesn't show a new one.

image
Not sure if this is a local machine problem. Could you maybe take a look at it and make sure it's working on your side? Thank you!

File: notifications/templates/notifications/view_notifications.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+{% extends 'events/base.html' %}
+
+{% block content %}
+    <h2>All Notifications</h2>
+    {% for notification in notifications %}
+        <div>
+            <p>{{ notification.message }}</p>
+            <small>{{ notification.timestamp }}</small>
+            {% if not notification.is_read %}
+                <a href="{% url 'notifications:mark-notification-as-read' notification.id %}">Mark as Read</a>
```
Comment (WenjunYuAnny) [inline notifications/templates/notifications/view_notifications.html:11]: The code looks correct, but when testing locally, the "mark as read" link doesn't show up even when the message first appears. Could you also take a look at it? Thank you!
image

**Comment by revindsilva26** [comment]
Gonna start fresh with the notifications page.

**Comment by zackaidja** [comment]
It makes no sense to me that this code won't be used, when a good chunk of it is working and usable

**Review by Dhiren-NYU — APPROVED** [review]

**Comment by revindsilva26** [comment]
Completed and implemented differently

**Pull request closed without merging**